best-practices/rspec.md

```diff
@@ -0,0 +1,30 @@
+## Rspec style guide
+
+- First #describe what you are doing
+- Then establish the #context
+- #it only expects one thing (within reason)
+- Prefer explicitness - #it #its #specify #subject are nice shortcuts but
+consider whether they are sacrificing readability
+- Having excessive setup (fixtures, database entities, doubles etc.) can cause false-positives. Thus, set up preconditions as close to the test as possible (usually within the context block) and tear them down afterwards. This is really important when working with time for example.
+- Unit tests are fast and effective to work with when doing TDD. These are isolated tests, typically
+for a single class, and should use stubs and spies via test doubles to replace anything that is
```

Comment:
I get a bit of this too. It makes me stop and think, usually, and decide whether I'm really doing that, or whether it means I'm just handing off functionality to something external well. Sometimes it means that my code is simple enough that there's not really much else to test.